contracts/Oracle.sol

```diff
@@ -54,8 +60,18 @@ contract Oracle {
         ) = UniswapV2OracleLibrary.currentCumulativePrices(address(pair));
         uint32 timeElapsed = blockTimestamp - blockTimestampLast; // overflow is desired
 
-        if (timeElapsed < PERIOD) {
-            // doesn't need to be updated, since a minimum period is not elapsed yet
+        if (block.timestamp < startTime) {
+            if (timeElapsed < PERIOD) {
+                return;
+            }
+        } else {
+            if (operator() != msg.sender) {
+                return;
+            }
+        }
```

Comment:
should this be a modifier? looks like it makes the function call irrelevant if it's not the right entity calling the contract / function.

Reply:
That's a good idea. It might be like this
```
modifier checkOperator {
    if (block.timestamp >= startTime && operator() != msg.sender) {
        return;
    }
    _;
}
```

Reply:
I implemented some codes based on your idea. So please review it. Thanks :)